Updated docs on running single tests (#12801)
GoogleCloudPlatform/magic-modules: docs/content/test/run-tests.md

1. Run acceptance tests for only modified resources. (Full test runs can take over 9 hours.) See [Go's documentation](https://pkg.go.dev/cmd/go#hdr-Testing_flags) for more information about `-run` and other flags.

    ```bash

>     make testacc TEST=./google/services/container TESTARGS='-run=TestAccContainerNodePool_basic$$'

Reviewer: I’d suggest also including an example for running multiple tests. For instance, you could explicitly state that the above example is for running single tests, and then add something like:

To run multiple tests with the `TestAccContainerNodePool` prefix:
```bash
    make testacc TEST=./google/services/container TESTARGS='-run=TestAccContainerNodePool'
```

Author: Thanks. That's what I was thinking as well, but because the wording was kind of narrow, and referred to the upstream docs, I wasn't totally sure. I can add something here.

    ```bash
    make testacc TEST=./google-beta/services/container TESTARGS='-run=TestAccContainerNodePool'
    ```

    To run all tests matching, e.g., `TestAccContainerNodePool*`, omit the trailing `$$`:

    ```bash

>     make testacc TEST=./google/services/container TESTARGS='-run=TestAccContainerNodePool'

Author: oops, missed that copy/pasta. fixed now.

    TF_LOG=DEBUG TF_LOG_PATH=output.log TF_CLI_CONFIG_FILE="$HOME/tf-dev-override.tfrc" terraform apply
    ```

### Run Tests with VCR Locally

> Running tests in "replaying" mode locally can sometimes be useful. In particular, it can allow you to test more quickly, cheaply, and without spinning up real infrastructure, once you've got an initial recording.
> 
> It can also be helpful for debugging tests that seem to work locally, but fail in CI in replaying mode.
>    
> Make sure `$VCR_PATH` is set, and points to a local directory which exists, and set `VCR_MODE` appropriately.

Reviewer: ```suggestion
VCR tests record HTTP request/response interactions in cassettes and replay them in future runs without calling the real API.

Running tests in "REPLAYING" mode locally can sometimes be useful. In particular, it can allow you to test more quickly, cheaply, and without spinning up real infrastructure, once you've got an initial recording.

It can also be helpful for debugging tests that seem to work locally, but fail in CI in replaying mode.
   
VCR is controlled via two variables:
- `VCR_MODE`: `REPLAYING` or `RECORDING` mode 
- `VCR_PATH`: Path where recorded cassettes are stored. 

Ensure both variables are configured to properly trigger VCR tests locally.
```

Reviewer: I added a bit more details. Let me know your thoughts!

> If you don't already have an existing cassette that's up to date, first do a run in "recording" mode:

Reviewer: ```suggestion
If you don't already have an existing cassette that's up to date, first do a run in `RECORDING` mode:
```

nit - ensure users can copy and run it directly

Author: thanks - I had it that way originally and thought maybe it would look too distracting, but good with it this way.

   
```bash
VCR_PATH=$HOME/.vcr/ VCR_MODE=RECORDING  make testacc TEST=./google/services/alloydb TESTARGS='-run=TestAccContainerNodePool_basic$$'
```

> Now run the same test again in "replaying" mode:

Reviewer: ```suggestion
Now run the same test again in `REPLAYING` mode:
```
